pkg/kclient/ingress.go

```diff
@@ -3,15 +3,41 @@ package kclient
 import (
 	"context"
 	"fmt"
-
+	"github.com/openshift/odo/pkg/unions"
 	"github.com/pkg/errors"
 
 	extensionsv1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CreateIngress creates an ingress object for the given service and with the given labels
-func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingress, error) {
+func (c *Client) CreateIngress(ingress unions.KubernetesIngress) (*unions.KubernetesIngress, error) {
+	var err error
+	if !ingress.IsGenerated() {
+		return nil, fmt.Errorf("please pass a generated ingress")
```

Comment:
As this is a problem at development time and not usage time, it should be better to have an assertion here (panic?).

Reply:
Also, please document in the function doc that the ingress param should be a *generated* ingress

Reply:
Updated documentation and adding panic as well